SU2_DEF/src/drivers/CDeformationDriver.cpp

```diff
@@ -0,0 +1,804 @@
+    if (rank == MASTER_NODE) cout << "Deleted CNumerics container." << endl;
+
+    for (iZone = 0; iZone < nZone; iZone++) {
+      delete solver_container[iZone];
+    }
+    delete [] solver_container;
+    if (rank == MASTER_NODE) cout << "Deleted CSolver container." << endl;
+
+    if (geometry_container != nullptr) {
+      for (iZone = 0; iZone < nZone; iZone++) {
+        delete geometry_container[iZone];
+      }
+      delete [] geometry_container;
+    }
+    if (rank == MASTER_NODE) cout << "Deleted CGeometry container." << endl;
+
+    if (surface_movement != nullptr) {
+      for (iZone = 0; iZone < nZone; iZone++) {
+        delete surface_movement[iZone];
+      }
+      delete [] surface_movement;
+    }
+    if (rank == MASTER_NODE) cout << "Deleted CSurfaceMovement class." << endl;
+
+    if (grid_movement != nullptr) {
+      for (iZone = 0; iZone < nZone; iZone++) {
+        delete grid_movement[iZone];
+      }
+      delete [] grid_movement;
+    }
+    if (rank == MASTER_NODE) cout << "Deleted CVolumetricMovement class." << endl;
+
+    if (config_container != nullptr) {
+      for (iZone = 0; iZone < nZone; iZone++) {
+        delete config_container[iZone];
+      }
+      delete [] config_container;
+    }
+    if (rank == MASTER_NODE) cout << "Deleted CConfig container." << endl;
+
+    if (output_container != nullptr) {
+      for (iZone = 0; iZone < nZone; iZone++) {
+        delete output_container[iZone];
+      }
+      delete [] output_container;
+    }
+    if (rank == MASTER_NODE) cout << "Deleted COutput class." << endl;
+
+    /*--- Exit the solver cleanly ---*/
+
+    if (rank == MASTER_NODE)
+      cout << endl << "------------------------- Exit Success (SU2_DEF) ------------------------" << endl << endl;
+}
+
+vector<string> CDeformationDriver::GetAllDeformMeshMarkersTag() const {
+
+  vector<string> interfaceBoundariesTagList;
+  unsigned short iMarker, nBoundariesMarker;
+  string Marker_Tag;
+
+  nBoundariesMarker = config_container[ZONE_0]->GetnMarker_Deform_Mesh();
+  interfaceBoundariesTagList.resize(nBoundariesMarker);
+
+  for(iMarker=0; iMarker < nBoundariesMarker; iMarker++){
+    Marker_Tag = config_container[ZONE_0]->GetMarker_Deform_Mesh_TagBound(iMarker);
+    interfaceBoundariesTagList[iMarker] = Marker_Tag;
+  }
+
+  return interfaceBoundariesTagList;
+}
+
+map<string, int> CDeformationDriver::GetAllBoundaryMarkers() const {
+
+  map<string, int>  allBoundariesMap;
+  unsigned short iMarker, nBoundaryMarkers;
+  string Marker_Tag;
+
+  nBoundaryMarkers = config_container[ZONE_0]->GetnMarker_All();
+
+  for(iMarker=0; iMarker < nBoundaryMarkers; iMarker++){
+    Marker_Tag = config_container[ZONE_0]->GetMarker_All_TagBound(iMarker);
+    allBoundariesMap[Marker_Tag] = iMarker;
+  }
+
+  return allBoundariesMap;
+}
+
+map<string, string> CDeformationDriver::GetAllBoundaryMarkersType() const {
+
+  map<string, string> allBoundariesTypeMap;
+  unsigned short iMarker, KindBC;
```

Comment:
I see what you mean with the duplication of python wrapper methods now... This needs a better solution, either this class needs to be derived from CDriver, or this class and CDriver need to share a new base class that provides this functions.
You can move these new files to SU2_CFD/src/drivers, if it makes that more natural.
Please give this some thought because duplication causes very stubborn bugs to remove.